plugins/inventory/kubevirt.py

```diff
@@ -314,18 +332,26 @@ def get_port_from_service(service: Dict) -> Optional[str]:
         get_port_from_service extracts the port to be used from the
         passed in service.
         """
-        # LoadBalancer services use the port attribute
-        if service["spec"]["type"] == TYPE_LOADBALANCER:
-            return service["spec"]["ports"][0]["port"]
+        ports = service.get("spec", {}).get("ports", [])
+        if not ports:
+            return None
 
-        # NodePort services use the nodePort attribute
-        if service["spec"]["type"] == TYPE_NODEPORT:
-            return service["spec"]["ports"][0]["nodePort"]
+        service_type = service.get("spec", {}).get("type")
+        if service_type == TYPE_LOADBALANCER:
+            # LoadBalancer services use the port attribute
+            return ports[0].get("port")
```

Comment:
What if the service contains multiple ports?

Reply:
Then we'd still have to pick one of them. I'd say first port is acceptable in that case.

Reply:
What if I would prefer the second one? :D

Reply:
Then I would ask you why you are adding two ports at all.